### PR TITLE
Submit form  pull request

### PR DESCRIPTION
**Pull request by AleWaweru**
- [ ] Created a method called current_user in  ApplicationController to make current user data available to all controllers.

- [ ] Created forms to perform the following functions:

- [ ] Created a Post on behalf of the current_user

- [ ] Created a comment on behalf of the current_user

- [ ] Created a function to Allow Users  add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      redirect_to user_post_path(user_id: @post.author.id, id: @post.id), notice: 'Comment added successfully'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
```
Comment (DuaneDave) [inline app/controllers/comments_controller.rb:22]: I have looked through the implementation of the form and everything seems good to me but one issue I faced is with testing the forms to ensure there are no errors. When I attempt to run migrations, I get some errors in the console that have to do with renaming a column that doesn't exist. Please take a look at the screenshot below in order to see the error so you can tackle it in order for a reviewer to be able to test the project.

image

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @AleWaweru 👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Views are created for both the comments and  post respectively

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there @AleWaweru👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ App matches requirements
✔️ No linter errors

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
- [ ] Great work attempting to fix the migration errors the previous reviewer mentioned, however, I am still getting those errors and still unable to test the app. Kindly take a look at the affected file in the screenshot below to trace the issue.

Screenshot 2023-09-02 at 23 29 38


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmetbozaci** [comment]
### Hello @AleWaweru 
### _This is an additional review as a result of the second opinion request._ 

- I am not getting any error after `rails db:migrate` command

image
**_As this was the only reason your project was not approved it will be approved as soon as possible._**

 _Please note that since we will approve it manually so this could **take a few hours**_


___



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**